csrc/scheduler/mma_utils.cpp

```diff
@@ -281,6 +281,143 @@ void scheduleWarpTileWithReduction(TensorView* tv, MatMulTileOptions tile) {
   // [Kwo Mwo Nwo Mw Nw Mi Ni Ki]
 }
 
+void scheduleWarpTile(
+    TensorView* tv,
+    MatMulTileOptions tile,
+    const std::vector<MatmulDimRole>& merged_roles) {
+  // Tiles are always given in [M, N, K]
+  auto cta_tile = tile.cta_tile;
+  auto warp_tile = tile.warp_tile;
+  auto instruction_tile = tile.instruction_tile;
+
+  // Do not split K dimension of CTA tile into multiple warp tiles
+  NVF_CHECK(
+      cta_tile.k == warp_tile.k,
+      "CTA tile and warp tile must have same K dimension");
+
+  // Find _inner_ tile dimensions present in tv
+  std::vector<int64_t> inner_dims;
+  std::vector<int64_t> required_inner_dim_sizes;
+  int64_t m_dim = -1, n_dim = -1, k_dim = -1;
```

Comment:
Please include your improvements in https://github.com/NVIDIA/Fuser/pull/2992, such as https://github.com/NVIDIA/Fuser/pull/2992/commits/5c19d4edee3be8ec9eedeaf0b53f97e62f719f21.

Reply:
Ah of course. I'll re-request a review when it's ready again.